### ISSUE TITLE
Add a command for generating a legal boilerplate page

### ISSUE THREAD

**Issue by 23Skidoo** [haskell/cabal#4887]
As @hvr mentioned in #4883, a lot of libraries `cabal-install` depends on have the following (or similar) standard requirement:

> Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.

Since we provide binary downloads of `cabal-install`, we should somehow comply with that. I think that the easiest way would be to add a "Legal" appendix to the manual listing the licenses of all of our dependencies. There should also be a way to generate such a page from a `.cabal` file in HTML or/and ReST format -- and since it's generally useful, this functionality should be part of `cabal-install`.

**Comment by hvr** [comment]
It's a bit tricky to get `cabal-install` to automatically generate a license manifest of its dependencies license files and have it included in the executable without some hefty custom `Setup.hs` which needs to interact with the package db (and also needs to handle inplace packages).

Maybe it's easier to have a license manifest generated as a text-file (think a command `cabal gen-manifest [FLAGS] <component(s)>` modulo bikeshed; I think that's what you're suggesting?), like e.g.
https://gist.github.com/ekmett/34024e26479763d77bc3e1ecaa24b819 and place it close to the respective `cabal` binaries; it'd be generally useful to have an exact manifest of all the artifacts and their versions that were compiled into the `cabal` executable.

See also https://github.com/haskell/cabal/issues/879#issuecomment-336367289

**Comment by 23Skidoo** [comment]
Yep, generating a text (reST/Markdown/...) file is what I had in mind.
